src/main/java/core/mvc/tobe/AnnotationHandlerMapping.java

```diff
@@ -1,27 +1,90 @@
 package core.mvc.tobe;
 
 import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import core.annotation.web.Controller;
+import core.annotation.web.RequestMapping;
 import core.annotation.web.RequestMethod;
+import org.reflections.Reflections;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import javax.servlet.http.HttpServletRequest;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 public class AnnotationHandlerMapping {
+    private static final Logger logger = LoggerFactory.getLogger(AnnotationHandlerMapping.class);
     private Object[] basePackage;
-
     private Map<HandlerKey, HandlerExecution> handlerExecutions = Maps.newHashMap();
 
     public AnnotationHandlerMapping(Object... basePackage) {
         this.basePackage = basePackage;
     }
 
+    @SuppressWarnings("unchecked")
     public void initialize() {
-
+        Reflections reflections = new Reflections(basePackage);
+        Set<Class<?>> preInstantiateClasses = getTypesAnnotatedWith(reflections, Controller.class);
+        for (Class<?> preInstantiateClass : preInstantiateClasses) {
+            List<Method> methods = findRequestMappingMethods(preInstantiateClass);
+            try {
+                putHandlerExecutions(preInstantiateClass.newInstance(), methods);
+            } catch (InstantiationException | IllegalAccessException e) {
+                logger.error(e.getMessage());
+            }
+        }
     }
 
     public HandlerExecution getHandler(HttpServletRequest request) {
         String requestUri = request.getRequestURI();
-        RequestMethod rm = RequestMethod.valueOf(request.getMethod().toUpperCase());
-        return handlerExecutions.get(new HandlerKey(requestUri, rm));
+        RequestMethod requestMethod = RequestMethod.valueOf(request.getMethod().toUpperCase());
+        HandlerKey handlerKey = handlerExecutions.keySet().stream()
+                .filter(h -> h.isMatchKey(requestUri, requestMethod))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Handler를 찾을 수 없습니다."));
+
+        return handlerExecutions.get(handlerKey);
```

Comment:
스트림으로 바로 리턴할 수 있을 것 같아요! :)

```suggestion
        return handlerExecutions.keySet().stream()
                .filter(h -> h.isMatchKey(requestUri, requestMethod))
                .findFirst()
                .map(it -> handlerExecutions.get(it))
                .orElseThrow(() -> new IllegalArgumentException("Handler를 찾을 수 없습니다."));
```